tket/tests/test_AASRoute.cpp

```diff
@@ -0,0 +1,429 @@
+#include <algorithm>
+#include <catch2/catch.hpp>
+
+#include "Circuit/Circuit.hpp"
+#include "Mapping/AASRoute.hpp"
+#include "Mapping/MappingManager.hpp"
+#include "OpType/OpType.hpp"
+#include "OpType/OpTypeFunctions.hpp"
+#include "Predicates/CompilationUnit.hpp"
+#include "Predicates/CompilerPass.hpp"
+#include "Predicates/PassGenerators.hpp"
+#include "Predicates/PassLibrary.hpp"
+#include "Simulation/CircuitSimulator.hpp"
+#include "Simulation/ComparisonFunctions.hpp"
+#include "Transformations/ContextualReduction.hpp"
+#include "testutil.hpp"
+
+namespace tket {
+SCENARIO("Test aas route in RV3") {
+  std::vector<Node> nodes = {
+      Node("test_node", 0), Node("test_node", 1), Node("test_node", 2),
+      Node("node_test", 3), Node("node_test", 4), Node("node_test", 5),
+      Node("test_node", 6), Node("node_test", 7), Node("node_test", 8),
+      Node("node_test", 9), Node("node_test", 10)};
+
+  Architecture architecture(
+      {{nodes[0], nodes[1]},
+       {nodes[1], nodes[2]},
+       {nodes[2], nodes[3]},
+       {nodes[3], nodes[4]},
+       {nodes[2], nodes[5]},
+       {nodes[5], nodes[6]},
+       {nodes[4], nodes[7]},
+       {nodes[7], nodes[8]},
+       {nodes[8], nodes[9]},
+       {nodes[9], nodes[10]}});
+  ArchitecturePtr shared_arc = std::make_shared<Architecture>(architecture);
+
+  GIVEN("test check_method in AASRouteRoutingMethod I") {
+    Circuit circ(11);
+    std::vector<Qubit> qubits = circ.all_qubits();
+
+    circ.add_op<UnitID>(OpType::CX, {qubits[0], qubits[4]});
+    circ.add_op<UnitID>(OpType::CX, {qubits[6], qubits[7]});
+    circ.add_op<UnitID>(OpType::CX, {qubits[1], qubits[10]});
+    circ.add_op<UnitID>(OpType::CX, {qubits[8], qubits[5]});
+    circ.add_op<UnitID>(OpType::CX, {qubits[3], qubits[9]});
+
+    circ.add_op<UnitID>(OpType::CX, {qubits[1], qubits[5]});
+    circ.add_op<UnitID>(OpType::CX, {qubits[3], qubits[9]});
+    circ.add_op<UnitID>(OpType::CX, {qubits[10], qubits[0]});
+    circ.add_op<UnitID>(OpType::CX, {qubits[6], qubits[0]});
+
+    std::map<UnitID, UnitID> rename_map = {
+        {qubits[0], nodes[0]}, {qubits[1], nodes[1]},  {qubits[2], nodes[2]},
+        {qubits[3], nodes[3]}, {qubits[4], nodes[4]},  {qubits[5], nodes[5]},
+        {qubits[6], nodes[6]}, {qubits[7], nodes[7]},  {qubits[8], nodes[8]},
+        {qubits[9], nodes[9]}, {qubits[10], nodes[10]}};
+
+    circ.rename_units(rename_map);
+
+    std::shared_ptr<MappingFrontier> mf =
+        std::make_shared<MappingFrontier>(circ);
+
+    AASRouteRoutingMethod aasrm(aas::CNotSynthType::Rec, 1);
+
+    REQUIRE(!aasrm.check_method(mf, shared_arc));
+  }
+  GIVEN("test check_method in AASRouteRoutingMethod II") {
```

Comment:
is there a fundamental difference between I and II? May be worth making the titles more descriptive